file/data.go

```diff
@@ -0,0 +1,193 @@
+package file
+
+// Dotfile tracks files by writing to a json file in the users home direcory.
+// This file provides functions for saving and retreiving json data.
+//
+// Example generated json:
+// {
+//   "bashrc": {
+//     "path": "~/.bashrc"
+//     "current": "451de414632e08c0ca3adf7a473b15f37c1b2e60"
+//     "commits": [{
+//       "hash":"451de414632e08c0ca3adf7a473b15f37c1b2e60",
+//       "timestamp":"1558896245",
+// 	 "message": "add aliases for dotfile"
+//    }],
+//  },
+//   "emacs": {
+//     "path": "~/.emacs.d/init.el",
+//     "commits": [{
+//       "hash":"8f94c7720a648af9cf9dab33e7f297d28b8bf7cd",
+//       "timestamp":"1558896290",
+//       "message": "add bindings for dotfile"
+//  }]
+// }
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
+
+// Data provides methods for reading and writing to a data json file.
+// All exported fields should be set.
+type Data struct {
```

Comment:
I don't like `Data` as a name. Maybe `Backend` or `Storage`?

Reply:
I didn't like it either. I'll refactor to Storage.